src/main/kotlin/com/terraformation/backend/documentproducer/db/VariableValueStore.kt

```diff
@@ -881,4 +883,19 @@ class VariableValueStore(
           QuestionsDeliverableSubmittedEvent(deliverableId, projectId, highestValuesByVariables))
     }
   }
+
+  private fun updateStatus(
+      projectId: ProjectId,
+      values: List<ExistingValue>,
+  ) {
+    val variableRows =
+        values
+            .map { it.variableId }
+            .toSet()
+            .associateWith { variablesDao.fetchOneById(it) ?: throw VariableNotFoundException(it) }
```

Comment:
I don't think this line is needed, you should be able to just pass the variable ID along in the event without fetching the full variable.

Reply:
I kept that line to check if the variable has a deliverableId. I think we don't need to change the status if it doesn't have a deliverable associated

Reply:
Oh I see! Makes sense